Unable to build with Rust enabled in closed environment

**Issue Description**

COPR builds systems as well as RHEL do not allow access to the network to do the build.  Enablign Rust currently still needs to do this and it is breaking the build...

`
SLAPD_HEADER_DIR=/builddir/build/BUILD/389-ds-base-2.0.0.0.20201103git5b0cbddc4/ \
	cargo rustc  --manifest-path=./src/librnsslapd/Cargo.toml \
	--release --verbose -- -C debuginfo=2
    Updating crates.io index
warning: spurious network error (2 tries remaining): failed to resolve address for github.com: Temporary failure in name resolution; class=Net (12)
warning: spurious network error (1 tries remaining): failed to resolve address for github.com: Temporary failure in name resolution; class=Net (12)
error: failed to get `libc` as a dependency of package `librslapd v0.1.0 (/builddir/build/BUILD/389-ds-base-2.0.0.0.20201103git5b0cbddc4/src/librslapd)`
Caused by:
  failed to fetch `https://github.com/rust-lang/crates.io-index`
Caused by:
  network failure seems to have happened
  if a proxy or similar is necessary `net.git-fetch-with-cli` may help here
  https://doc.rust-lang.org/cargo/reference/config.html#netgit-fetch-with-cli
Caused by:
  failed to resolve address for github.com: Temporary failure in name resolution; class=Net (12)
make: *** [Makefile:13775: /builddir/build/BUILD/389-ds-base-2.0.0.0.20201103git5b0cbddc4/rs/release/librslapd.a] Error 101
make: *** Waiting for unfinished jobs....
    Updating crates.io index
warning: spurious network error (2 tries remaining): failed to resolve address for github.com: Temporary failure in name resolution; class=Net (12)
warning: spurious network error (1 tries remaining): failed to resolve address for github.com: Temporary failure in name resolution; class=Net (12)
error: failed to get `libc` as a dependency of package `librslapd v0.1.0 (/builddir/build/BUILD/389-ds-base-2.0.0.0.20201103git5b0cbddc4/src/librslapd)`
Caused by:
  failed to fetch `https://github.com/rust-lang/crates.io-index`
Caused by:
  network failure seems to have happened
  if a proxy or similar is necessary `net.git-fetch-with-cli` may help here
  https://doc.rust-lang.org/cargo/reference/config.html#netgit-fetch-with-cli
Caused by:
  failed to resolve address for github.com: Temporary failure in name resolution; class=Net (12)
make: *** [Makefile:13786: /builddir/build/BUILD/389-ds-base-2.0.0.0.20201103git5b0cbddc4/rs/release/librnsslapd.a] Error 101
`

So we need to do something like what we do with Cockpit where we pull everything in and build it first when generating an srpm.

This is still failing for me with this flag set (different error though), when I generate an srpm and send it to copr:

```
...
error: failed to get `concread` as a dependency of package `librslapd v0.1.0 (/builddir/build/BUILD/389-ds-base-2.0.0.0.20201104gitfc8a94943/src/librslapd)`
Caused by:
  failed to load source for dependency `concread`
Caused by:
  Unable to update registry `https://github.com/rust-lang/crates.io-index`
Caused by:
  failed to update replaced source registry `https://github.com/rust-lang/crates.io-index`
Caused by:
  failed to read root of directory source: /builddir/build/BUILD/389-ds-base-2.0.0.0.20201104gitfc8a94943/vendor
Caused by:
  No such file or directory (os error 2)
make: *** [Makefile:13782: /builddir/build/BUILD/389-ds-base-2.0.0.0.20201104gitfc8a94943/rs/release/librnsslapd.a] Error 101
error: Bad exit status from /var/tmp/rpm-tmp.6Nmhbt (%build)
    Bad exit status from /var/tmp/rpm-tmp.6Nmhbt (%build)
```

--- a/rpm.mk
+++ b/rpm.mk
@@ -25,7 +25,7 @@ TSAN_ON = 0
 # Undefined Behaviour Sanitizer
 UBSAN_ON = 0
 
-RUST_ON = 0
+RUST_ON = 1
 
 COCKPIT_ON = 1
 
@@ -39,6 +39,7 @@ update-cargo-dependencies:
 	cargo update --manifest-path=./src/Cargo.toml
 
 download-cargo-dependencies:
+	cargo update --manifest-path=./src/Cargo.toml
 	cargo vendor --manifest-path=./src/Cargo.toml
 	cargo fetch --manifest-path=./src/Cargo.toml
 	tar -czf vendor.tar.gz vendor
@@ -120,7 +121,7 @@ rpmbuildprep:
 		cp dist/sources/$(JEMALLOC_TARBALL) $(RPMBUILD)/SOURCES/ ; \
 	fi
 
-srpms: rpmroot srpmdistdir tarballs rpmbuildprep
+srpms: rpmroot srpmdistdir download-cargo-dependencies tarballs rpmbuildprep 
 	rpmbuild --define "_topdir $(RPMBUILD)" -bs $(RPMBUILD)/SPECS/$(PACKAGE).spec
 	cp $(RPMBUILD)/SRPMS/$(RPM_NAME_VERSION)*.src.rpm dist/srpms/
 	rm -rf $(RPMBUILD)

--- a/rpm/389-ds-base.spec.in
+++ b/rpm/389-ds-base.spec.in
@@ -337,7 +337,7 @@ UBSAN_FLAGS="--enable-ubsan --enable-debug"
 %endif
 
 %if %{use_rust}
-RUST_FLAGS="--enable-rust"
+RUST_FLAGS="--enable-rust --enable-rust-offline"
 %endif
 
 %if !%{use_cockpit}

